.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,138 @@
+# This reusable workflow encapsulates an end to end workflow for building and
+# pushing a container image for a MozCloud Service. It:
+#   - runs a "preflight" script to do things like create a `version.json` file
+#   - builds the image
+#   - provides an entrypoint to run verification steps on the image (e.g. load tests)
+#   - pushes the image
+#
+# For details on the workflow inputs passed to docker-build and docker-push, see the 
+# documentation for those actions:
+#  
+# https://github.com/mozilla-it/deploy-actions/tree/main/docker-build
+# https://github.com/mozilla-it/deploy-actions/tree/main/docker-push
+
+name: Docker Build and Push
+
+on:
+  workflow_call:
+    inputs:
+      image_name:
+        required: true
+        type: string
+      gar_name:
+        required: true
+        type: string
+      project_id:
+        required: true
+        type: string
+      workload_identity_pool_project_number:
+        required: true
+        type: string
+      preflight_script:
+        type: string
+        required: false
+        default: |
+          IMAGE_BUILD_CONTEXT_ABSPATH=$(realpath "$IMAGE_BUILD_CONTEXT")
+
+          printf '{"commit":"%s","version":"%s","source":"%s","build":"%s"}\n' \
+          "$GITHUB_SHA" \
+          "$GITHUB_REF_NAME" \
+          "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
+          "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" > "$IMAGE_BUILD_CONTEXT_ABSPATH/version.json"
+      verify_script:
```

Comment:
would it make sense to rename this to `postflight_script` to keep the names consistent?